client/cgo/coniks.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import "C"
+
+import (
+	"bytes"
+	"unsafe"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+// main is required to build a shared library, but does nothing
+func main() {}
+
+//export cgoEd25519SignatureVerify
+func cgoEd25519SignatureVerify(pk unsafe.Pointer, pkSize C.int,
+	message unsafe.Pointer, size C.int,
+	sig unsafe.Pointer, sigSize C.int) C.int {
+	if int(pkSize) < crypto.PublicKeySize {
+		return 0
+	}
+	pkBytes := C.GoBytes(pk, pkSize)
+	messageBytes := C.GoBytes(message, size)
+	sigBytes := C.GoBytes(sig, sigSize)
+	key := crypto.VerifKey(pkBytes)
+	if key.Verify(messageBytes, sigBytes) {
+		return 1
+	}
+	return 0
+}
+
+//export cgoVrfVerify
+func cgoVrfVerify(pk unsafe.Pointer, pkSize C.int,
+	m unsafe.Pointer, size C.int,
+	index unsafe.Pointer, indexSize C.int,
+	proof unsafe.Pointer, proofSize C.int) C.int {
+	if int(pkSize) < vrf.PublicKeySize ||
+		int(indexSize) < vrf.Size ||
+		int(proofSize) < vrf.ProofSize {
+		return 0
+	}
+	pkBytes := C.GoBytes(pk, pkSize)
+	mBytes := C.GoBytes(m, size)
+	vrfBytes := C.GoBytes(index, indexSize)
+	proofBytes := C.GoBytes(proof, proofSize)
+	if vrf.Verify(pkBytes, mBytes, vrfBytes, proofBytes) {
```

Comment:
> I gave it a try in a786a02. What do you think?

Not bad.  You'll probably have a naming clash for `SecretKey / VerifKey` when you flatten things out.

> Should we have some APIs named: crypto.SignatureGenerateKey, crypto.VrfGenerateKey and key types: SignatureVerifKey, VrfVerifKey, then I'd call pk.Verify. What do you think?

I'm not crazy about calling these `Secret / Verif`.  I prefer the conventional`Public / Private` pair.  Also, the verb should come first, as in `GenerateSignKey`.  Maybe `PublicSignKey`, `PrivateSignKey`, `PublicVrfKey`, `PrivateVrfKey`?